core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -48,6 +54,41 @@ class NettyBlockTransferService(conf: SparkConf, securityManager: SecurityManage
   private[this] var clientFactory: TransportClientFactory = _
   private[this] var appId: String = _
 
+  override def metricsCollection(executorId: String, executorMetrics: ExecutorMetrics): Unit = {
+    logDebug("starting a new thread for metricsCollection()")
+    new Thread("metric thread for executor " + executorId) {
+      override def run() {
+        while (true) {
+          getNettyMemMetrics(executorMetrics)
+          Thread.sleep(2000)
```

Comment:
couldn't this just be done as part of the heartbeat?  Why do we need a separate thread for this?  Is it just so that we can collect the max usage at a finer resolution?

I think it makes sense to want a better resolution for the max memory, but I don't think this is the right way to do it.  There may be many metrics which we'd like to poll more frequently than the heartbeat.  I feel like for v1, we should jsut collect the metrics on the heartbeat, and then its a separate issue of enabling more rapid metric collection.

I'm not totally certain about this -- what do you think?